### ISSUE TITLE
synchronise final head state

### ISSUE THREAD

**Issue by alex-courtis** [alex-courtis/way-displays#22]
Start with eDP-1 only, plug in HDMI-A-1

```
I [15:11:46.301] way-displays version 1.3.1-SNAPSHOT
I [15:11:46.350]
I [15:11:46.351] Found configuration file: /home/alex/.dotfiles/config/way-displays/cfg.yaml
I [15:11:46.591]   Arrange in a COLUMN aligned at the RIGHT
I [15:11:46.591]   Order:
I [15:11:46.591]     AAAAA
I [15:11:46.591]     BBBBB
I [15:11:46.591]     CCCCC
I [15:11:46.592]   Auto scale: ON
I [15:11:46.592]   Scale:
I [15:11:46.599]     AAAAA: 2.000
I [15:11:46.599]     BBBBB: 2.000
I [15:11:46.599]     HDMI-A-1: 2.000
I [15:11:46.599]   Max preferred refresh:
I [15:11:46.599]     B
I [15:11:46.599]     C
I [15:11:46.599]     A
I [15:11:46.599]     HDMI-A-1
I [15:11:46.599]   Disabled:
I [15:11:46.600]     AAAAA
I [15:11:46.600]     BBBBB
I [15:11:46.600]   Laptop display prefix: eDPP
I [15:11:46.646]
I [15:11:46.646] eDP-1 Arrived:
I [15:11:46.646]   info:
I [15:11:46.646]     name:     'eDP-1'
I [15:11:46.646]     desc:     'Unknown 0x05EF 0x00000000 (eDP-1)'
I [15:11:46.646]     width:    310mm
I [15:11:46.646]     height:   170mm
I [15:11:46.651]     dpi:      212.45 @ 2560x1440
I [15:11:46.652]   current:
I [15:11:46.653]     scale:    2.250
I [15:11:46.653]     position: 142,0
I [15:11:46.653]     mode:     2560x1440@60Hz (preferred)
I [15:11:46.654]
I [15:11:46.654] handle_changes
I [15:11:46.655]
I [15:11:46.655] handle_changes dirty no pending
I [15:11:46.662]
I [15:11:46.662] handle_changes now pending
I [15:11:46.662]
I [15:11:46.662] eDP-1 Changing:
I [15:11:46.662]   from:
I [15:11:46.662]     scale:    2.250
I [15:11:46.662]     position: 142,0
I [15:11:46.662]     mode:     2560x1440@60Hz (preferred)
I [15:11:46.662]   to:
I [15:11:46.663]     position: 0,0
I [15:11:46.714]
I [15:11:46.715] Changes successful
I [15:11:46.716]
I [15:11:46.716] handle_changes
I [15:11:46.717]
I [15:11:46.717] handle_changes output manager not pending
I [15:11:48.557]
I [15:11:48.557] Monitoring lid device: /dev/input/event1
I [15:11:48.560]
I [15:11:48.560] handle_changes
I [15:11:48.560]
I [15:11:48.560] handle_changes output manager not pending
I [15:11:54.139]
I [15:11:54.139] HDMI-A-1 Arrived:
I [15:11:54.139]   info:
I [15:11:54.139]     name:     'HDMI-A-1'
I [15:11:54.139]     desc:     'Unknown ASUS XG32V 0x0000CF52 (HDMI-A-1)'
I [15:11:54.139]     width:    700mm
I [15:11:54.139]     height:   390mm
I [15:11:54.142]   current:
I [15:11:54.142]     scale:    2.000
I [15:11:54.142]     position: 0,640
I [15:11:54.142]     mode:     2560x1440@60Hz
I [15:11:54.142]
I [15:11:54.142] handle_changes
I [15:11:54.142]
I [15:11:54.142] handle_changes dirty no pending
I [15:11:54.143]
I [15:11:54.143] handle_changes now pending
I [15:11:54.143]
I [15:11:54.144] eDP-1 Changing:
I [15:11:54.144]   from:
I [15:11:54.144]     scale:    2.250
I [15:11:54.144]     position: 0,0
I [15:11:54.144]     mode:     2560x1440@60Hz (preferred)
I [15:11:54.144]   to:
I [15:11:54.144]     position: 142,0
I [15:11:54.250]
I [15:11:54.250] Changes successful
I [15:11:54.250]
I [15:11:54.250] handle_changes
I [15:11:54.250]
I [15:11:54.251] handle_changes output manager not pending
I [15:11:56.126]
I [15:11:56.126] HDMI-A-1 Departed:
I [15:11:56.126]     name:     'HDMI-A-1'
I [15:11:56.127]     desc:     'Unknown ASUS XG32V 0x0000CF52 (HDMI-A-1)'
I [15:11:56.127]
I [15:11:56.127] handle_changes
I [15:11:56.127]
I [15:11:56.127] handle_changes dirty no pending
I [15:11:56.127]
I [15:11:56.127] handle_changes now pending
I [15:11:56.127]
I [15:11:56.128] eDP-1 Changing:
I [15:11:56.128]   from:
I [15:11:56.128]     scale:    2.250
I [15:11:56.128]     position: 142,0
I [15:11:56.128]     mode:     2560x1440@60Hz (preferred)
I [15:11:56.128]   to:
I [15:11:56.128]     position: 0,0
I [15:11:56.586]
I [15:11:56.586] HDMI-A-1 Arrived:
I [15:11:56.586]   info:
I [15:11:56.586]     name:     'HDMI-A-1'
I [15:11:56.586]     desc:     'Unknown ASUS XG32V 0x0000CF52 (HDMI-A-1)'
I [15:11:56.586]     width:    700mm
I [15:11:56.586]     height:   390mm
I [15:11:56.589]   current:
I [15:11:56.590]     scale:    2.000
I [15:11:56.590]     position: 0,640
I [15:11:56.590]     mode:     2560x1440@60Hz
I [15:11:56.590]
I [15:11:56.590] handle_changes
I [15:11:56.590]
I [15:11:56.591] Changes cancelled, retrying 1/3
I [15:11:56.591]
I [15:11:56.591] handle_changes
I [15:11:56.591]
I [15:11:56.591] handle_changes dirty no pending
I [15:11:56.591]
I [15:11:56.591] handle_changes output manager not pending
```

**Comment by alex-courtis** [comment]
The head state is not properly tracked following a cancel or fail from zwlr_output_configuration_v1.

Explicitly track and do a deep compare of the head state, rather than clearing pending.

**Comment by alex-courtis** [comment]
Resolved via c72a922
